src/ess/polarization/base.py

```diff
@@ -353,67 +342,132 @@ def he3_opacity_from_cell_params(
+    """Time-dependent polarization function for a given cell."""
+
+    def __init__(self, C: sc.Variable, T1: sc.Variable):
+        self._C = C
+        self._T1 = T1
+
+    @property
+    def C(self) -> sc.Variable:
+        return self._C
+
+    @property
+    def T1(self) -> sc.Variable:
+        return self._T1
 
+    def __call__(self, time: sc.Variable) -> sc.Variable:
+        return self.C * sc.exp(-time / self.T1)
 
-def he3_polarization(
+
+@dataclass
+class He3TransmissionFunction(Generic[Cell]):
+    """Wavelength- and time-dependent transmission for a given cell."""
+
+    opacity_function: He3OpacityFunction[Cell]
+    polarization_function: He3PolarizationFunction[Cell]
+    transmission_empty_glass: He3TransmissionEmptyGlass[Cell]
+
+    def __call__(
+        self,
+        *,
+        time: sc.Variable,
+        wavelength: sc.Variable,
+        plus_minus: Literal['plus', 'minus'],
+    ) -> sc.Variable:
+        opacity = self.opacity_function(wavelength)
+        polarization = self.polarization_function(time)
+        if plus_minus == 'plus':
+            polarization -= 1.0
+        else:
+            polarization += 1.0
+        return self.transmission_empty_glass * sc.exp(opacity * polarization)
+
+
+def transmission_incoming_unpolarized(
+    *,
+    transmission_empty_glass: sc.Variable,
+    opacity: sc.Variable,
+    polarization: sc.Variable,
+) -> sc.Variable:
+    return transmission_empty_glass * sc.exp(-opacity) * sc.cosh(opacity * polarization)
+
+
+def get_he3_transmission_from_fit_to_direct_beam(
     direct_beam_no_cell: DirectBeamNoCell,
     direct_beam_polarized: He3DirectBeam[Cell, Polarized],
-    opacity: He3OpacityFunction[Cell],
-    filling_time: He3FillingTime[Cell],
+    opacity_function: He3OpacityFunction[Cell],
     transmission_empty_glass: He3TransmissionEmptyGlass[Cell],
-) -> He3Polarization[Cell]:
+) -> He3TransmissionFunction[Cell]:
     """
-    Fit time- and wavelength-dependent equation and return the fit param P(t).
+    Return the transmission function for a given cell.
+
+    This is composed from the opacity-function and the polarization-function.
+    The implementation fits a time- and wavelength-dependent equation and returns
+    the fitted T(t, lambda).
 
     DB_pol/DB = T_E * cosh(O(lambda)*P(t))*exp(-O(lambda))
     """
-    # Each time bin corresponds to a direct beam measurement. Take the mean for each
-    # but keep the time binning.
-    # time_up = direct_beam_up.bins.coords['time'].bins.mean()
-    # time_down = direct_beam_down.bins.coords['time'].bins.mean()
-    # results dims: spin state, wavelength, time
-    raise NotImplementedError()
-    return He3Polarization[Cell](1)
-
 
-def he3_transmission(
-    opacity: He3OpacityFunction[Cell],
-    polarization: He3Polarization[Cell],
-    transmission_empty_glass: He3TransmissionEmptyGlass[Cell],
-) -> He3Transmission[Cell]:
-    """
-    Transmission for a given cell.
+    def direct_beam_ratio(
```

Comment:
I still prefer the name `transmission` here, or maybe `expected_transmission`. The "ratio" here is (imo) the observed transmission.